### PR TITLE
gdb extension: include debuginfod-find (debuginfod-client.rpm) 

### PR DESCRIPTION
**Pull request by mvidner**
- https://bugzilla.suse.com/show_bug.cgi?id=1182649
- https://trello.com/c/oj8K4EPu

so that valgrind, part of the extension, can benefit from the server:
https://debuginfod.opensuse.org/

### CODE REVIEW

**Comment by mvidner** [comment]
Huh, this is a confusing error at build time:

[    4s] [465/962] cumulate debuginfod-client-0.183-2.1
[    4s] debuginfod-client-0.183-2.1           ########################################
...
[  317s] theme=openSUSE libdeps=root,gdb image=gdb src=root fs=squashfs disjunct=root bin/mk_image
[  317s] creating solv file...
[  317s] --- Building for openSUSE Tumbleweed (openSUSE-Tumbleweed) x86_64 [lib64], theme openSUSE
[  317s] --- Kernel: kernel-default (virtualbox-guest-kmp, xen-kmp, drm-kmp, reiserfs-kmp), vmlinuz-5.12.0-1-default, 5.12.0-1-default
[  317s] --- Source: /.build.binaries
[  317s] --- Driver Updates: /linux/suse/x86_64-tw
[  317s] adding template #1 >.*<
[  318s] we need package gdb [10.1-4.1.x86_64] {postin,preun}
[  318s] we need package valgrind [3.17.0-1.1.x86_64]
[  318s] we need package debuginfod-client [0.183-2.1.x86_64]
[  318s] we ignore package yast2-buildtools
[  318s] resolving package dependencies...
**[  318s] mk_image: package debuginfod-client does not exist**
[  318s]  in data/root/gdb.file_list line 18.
[  318s] make: *** [Makefile:190: gdb] Error 25

**Comment by mvidner** [comment]
@wfeldt do you have an idea what's wrong here? `grep -R 'does.not.exist'` only finds one comment in the repo, I con't find the responsible code.

**Comment by wfeldt** [comment]
Probably the OBS solver:

https://github.com/openSUSE/installation-images/blob/master/lib/ReadConfig.pm#L676-L723

you can try setting the environment variable `debug=solv` to see the full solver report.

**Comment by mvidner** [comment]
The build problem went away when I used a clean buildroot, so it looks like a caching issue.

I have added basic tests for `resolve_deps_libsolv`.

File: data/root/gdb.file_list [hunk 1]
```diff
@@ -3,6 +3,8 @@ TEMPLATE:
 
 gdb:
 ?valgrind: nodeps
+# helps valgrind download debuginfos
+?debuginfod-client:
```
Comment (mvidner) [inline data/root/gdb.file_list:7]: This is the one-liner feature,
the rest of the PR are tests added in an attempt to debug a transient build failure :rofl:

**Review by wfeldt — APPROVED** [review]
lgtm

**Comment by wfeldt** [comment]
Just one note: you didn't add `debuginfod-client` to the spec file, so the change won't be active.

**Comment by mvidner** [comment]
@wfeldt Oh, I was looking in `package/` and missed `obs/`, thanks for the pointer!

**Pull request merged**